hashmap.h

```diff
@@ -288,6 +300,34 @@ int hashmap_iterate(const struct hashmap_s *const m,
   return 0;
 }
 
+int hashmap_iterate_pairs(struct hashmap_s *const m,
+                    int (*f)(void *const, const char *, unsigned, void *const), void *const context) {
+  unsigned int i;
+  hashmap_element_s *p;
+  int r;
+
+  /* Linear probing */
+  for (i = 0; i < m->table_size; i++)
+    p=&m->data[i];
+    if (p->in_use) {
+      r=f(context, p->key, p->key_len, p->data);
+      if (0<r) {
+        /* early return */
+        return 1;
+      }else if (0>r) {
```

Comment:
what happens if `r == 0` ?

Reply:
0 was the fall-through condition of -1 that would continue iteration.  Now replaced by switch statement.